util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -395,7 +395,15 @@ private static String getData(Map<String, Object> obj, String key) {
     return null;
   }
 
-  public static byte[] getDataOrFile(final String data, final String file) throws IOException {
+  public byte[] getDataOrFileRelative(final String data, final String path) throws IOException {
+    String resolvedPath = path;
+    if (resolvedPath != null && this.file != null) {
+      resolvedPath = this.file.toPath().getParent().resolve(path).normalize().toString();
+    }
+    return getDataOrFile(data, resolvedPath);
+  }
+
+  private static byte[] getDataOrFile(final String data, final String file) throws IOException {
```

Comment:
how about keeping this `public static`for compatibility?

Reply:
I don't think anyone should really be using this function. If you feel strongly I'll switch it back to public, but I think people should really use the new function.